### ISSUE TITLE
need way to "lock" rows together so they can be moved as a unit to slide

### ISSUE THREAD

**Issue by benwade** [McGill-CSB/PHYLO#57]
need way to "lock" rows together so they can be moved as a unit to slide

**Comment by waldispuhl** [comment]
I believe this feature is already implemented (not documented though...) : Double click and move the mouse to do a region select. All blocks within the unit will move as one. 
Please, let us know if that's the feature you are requesting or if I misunderstood your request. I'll also try to add asap a description of this functionality in the tutorial.

**Comment by benwade** [comment]
Yes, thank you. That was the feature I wanted. If you can, you might think
of some way to make it semi-permanent, and visually accessible  For example
put a small "lock" icon between locked rows. Click once to lock, again to
unlock.

Thanks, Ben

 One day the King decided that he would force all his subjects to tell
the truth. A gallows was erected in front of the city gates. A herald
announced, "Whoever would enter the city must first answer the truth to
a question which will be put to him." Nasrudin was first in line. The
captain of the guard asked him, "Where are you going? Tell the truth
-- the alternative is death by hanging." "I am going," said Nasrudin,
"to be hanged on that gallows." "I don't believe you." "Very well, if
I have told a lie, then hang me!" "But that would make it the truth!"
"Exactly," said Nasrudin, "your truth."

**Comment by Durdge** [comment]
I would be extremely interested in an advanced, persistent locking feature (particularly for the advanced version).  
More specifically I recommend the following implementation:
1. Use the double click highlight option, but have a lock icon appear in the center-ish of the selection 
2. Selecting the center icon will toggle between locking and unlocking the selected block.  The icon toggles between an unlocked and locked icon appropriately.  
3. Once locked, outline the locked areas so that the line eats into the selection area (so you do not encroach on other areas and side-by-side locked areas appear distinct.  Once unlocked, this line disappears  
4. When locked, leave the selected boxes with the dimmed color.
5. Pushing other rows into the locked box will push the entire box
6. You cannot lock one cell
7. Locked areas remain locked until unlocked
8. You can have as many locked areas as you want

I'm sure it's hard to implement, but it would be extremely useful feature because once you know you got a section perfect, you don't have to worry about screwing it up.

**Comment by rasmus-storjohann** [comment]
It would also be nice to be able to shift all the sequences of one subtree together, so that they can be shifted relative to other subtrees. This could perhaps be done with click-drag on the tree node to the left. Alternatively, there could be a way to click on the node and automatically lock all the sequences for that subtree.

**Comment by waldispuhl** [comment]
We already have a mechanism to click and select a region of the alignment that can be moved at a block. Is it doing (at least partially) what you are looking for? Alternatively, we currently testing a system to lock rows separately.

**Comment by rasmus-storjohann** [comment]
Yes, it is doing partially what I'm looking for. The current implementation has poor discoverability. I locked tiles a couple of times by accident, didn't know what it was for and ignored it. Then I discovered the need for locking (I'm still a newbie at this game, played for less than two hours), but I was not able to use the feature until I'd read about it in the docs. The feature is good, but it needs to be easier to invoke (It also has some bad bugs right now.) I'm not sure since I'm still new at playing this, but I think moving the entire rows of all the species of a sub-tree is a common use case. In this case tying the locking mechanism to the corresponding node in the tree on the left will be, in my view, intuitive. You can even put a lock icon on the tree node, the node does not seem to have any user interactions associated with it currently.

**Comment by rasmus-storjohann** [comment]
Actually, here's a different idea for locking:

So the situation is that I've aligned two sets of sequences, and now I want to align those two sets with each other. In the naive way, I have to manipulate each sequence individually. With the current lock implementation, I have to use the rubbber band to indicate both which sequences and which blocks within those sequences I want to move, even though in most cases the sequences I'll want to work with is defined by the tree, so there is information available that can be used to make my life easier. 

Here's my suggestion: Have a checkbox on each tree node. When I check a node, then I can manipulate any sequence in that subtree, and all the sequences in that subtree move as one. You don't need the red semitransparent block for this, at most changing the colour on the relevant nodes and edges in the tree would be sufficient. Sequences in subtrees that are not checked move individually as normal. Multiple sibling subtrees can be checked concurrently, and moving any block within either subtree manipulates all suquences in the checked subtree only.

**Comment by Durdge** [comment]
It sounds like there's some utility behind what rasmus is saying, but my request is different.  My request does not involve moving or locking at the "tree" level.  

Yes, you can select a block and move it around.  My request is allow the user the ability to lock that block so nothing within that block moves AND you can proceed to move and define other blocks.  

The use case here is that you've solved a block and there's no foreseeable reason to break it up, so you lock it (and it stays locked).  This allows you to drag a cell or row or block into the locked block and have it NOT move the entire line and thus messing up your block (or in some cases much more).  In addition, this would allow you to try to find matches on a macro level (i.e. shifting 2 or 3 blocks around to try to find the best fit.
